### DISCUSSION TITLE
root's index.d.ts

### DISCUSSION THREAD

**Discussion by rpokrovskij** [floating-ui/floating-ui#1372] [Q&A]
What is a role of root's index.d.ts ? Who is the user of it ?

**Comment by FezVrasta** [comment]
Its required by TypeScript to know where to find the type definitions

**Reply by rpokrovskij** [reply]
Thank you for you answer and tools. I want to explain my question (first for me): there is also lib\index.d.ts. Why you do not use `types` attribute from `package.json` to point directly to it ?  It doesn't work?
P.S. Trying to understand how this beauty was constructed: to have three versions "vanilla", "typescript" and "flow" packed together. Never seem such power before. Is it yours idea? Where I can read about it?

**Reply by FezVrasta** [reply]
I don't use the package.json because TypeScript expects me to define the entry file in a `types` property, this is not fair towards other type systems.

If TypeScript allowed to use something like `types.typescript` I would gladly use it.

I'm sure there are other packages built to support both Flow and TS, I think this may be they only one which generates TS types automatically starting from Flow types though.
